### PR TITLE
feat: add Dart v1.0 developer preview docs

### PR DESCRIPTION
**Pull request by dshukertjr**
Adds a new tab on the [Dart docs](https://supabase.com/docs/reference/dart) where users can view `v0` and `v1-dev` documentation. 
Screen Shot 2022-09-08 at 14 06 51

### CODE REVIEW

File: apps/reference/_supabase_dart/generated/storage-from-upload.mdx [hunk 1]
```diff
@@ -80,8 +78,8 @@ The body of the file to be stored in the bucket.
   <div class="method-list-item-description">
 
 HTTP headers.
-`cacheControl`: string, the `Cache-Control: max-age=<seconds>` seconds value.
-`contentType`: string, the `Content-Type` header value. Should be specified if using a `fileBody` that is neither `Blob` nor `File` nor `FormData`, otherwise will default to `text/plain;charset=UTF-8`.
+`cacheControl`: string, the `Cache-Control: max-age=<seconds>` seconds value. <br />
+`contentType`: string, the `Content-Type` header value. The SDK will read the content type and set the header for you by default.<br />
```
Comment (dannykng) [inline apps/reference/_supabase_dart/generated/storage-from-upload.mdx:82]:
```suggestion
`contentType`: string, the `Content-Type` header value. By default, the SDK reads the content type and sets the header.<br />
```

Comment (dannykng) [inline apps/reference/_supabase_dart/generated/storage-from-upload.mdx:82]: The formatting here is kind of hard to parse:
Screen Shot 2022-09-08 at 12 23 26 PM

Can we make this a "properties" table (e.g., [signUp()](https://supabase.com/docs/reference/javascript/auth-signup#parameters))?
Screen Shot 2022-09-08 at 12 24 44 PM

**Review by dannykng — COMMENTED** [review]
Left a few comments and suggestions!

File: apps/reference/_supabase_dart/intro.mdx [hunk 2]
```diff
@@ -6,6 +6,14 @@ sidebar_label: Supabase Dart Library
 
 # Supabase Dart Library
 
+:::note
+
+You're viewing the Supabase docs for the next version of our library which is not yet released.
+
+Refer to the `v0` docs for a stable release.
```
[the remark this answers is not in the document]
Comment (dshukertjr) [reply apps/reference/_supabase_dart/intro.mdx:13]: Great point. supabase-flutter library is in a bit of a weird state right now, because there is the v0, which is the "stable" release with no breaking changes, and there is the v1.0 developer preview. 

Maybe we can say something like this instead?

```
You're viewing the Supabase docs for a developer preview version.

Refer to the `v0` docs for the previous release.
```

File: apps/reference/_supabase_dart/generated/storage-from-upload.mdx (hunk 1 above)
Comment (dshukertjr) [reply apps/reference/_supabase_dart/generated/storage-from-upload.mdx]: Sounds good! Let me do this now!

File: apps/reference/_supabase_dart/intro.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (dshukertjr) [reply apps/reference/_supabase_dart/intro.mdx:13]: Done!

File: apps/reference/_supabase_dart/generated/storage-from-upload.mdx (hunk 1 above)
Comment (dshukertjr) [reply apps/reference/_supabase_dart/generated/storage-from-upload.mdx]: Done!

File: apps/reference/_supabase_dart/generated/storage-from-upload.mdx [hunk 3]
```diff
@@ -78,11 +76,71 @@ The body of the file to be stored in the bucket.
...
+    <span className="method-list-item-label-badge false">
+      optional
+    </span>
+    <span className="method-list-item-validation">
+      <code>string</code>
+    </span>
+  </h4>
+  <div class="method-list-item-description">
+
+The Content-Type header value. The SDK will read the content type and set the header for you by default.
```
Comment (dannykng) [inline apps/reference/_supabase_dart/generated/storage-from-upload.mdx:117]:
```suggestion
The Content-Type header value. By default, the SDK reads the content type and sets the header.
```

**Review by dannykng — APPROVED** [review]
Just one thing that didn't carry over but LGTM otherwise!
